### ISSUE TITLE
Account ACM

### ISSUE THREAD

**Issue by matheusfrancisco** [lorenzodifuccia/safaribooks#160]
my safari login is by acm i would like to know if this code works for ACM? If it doesn't work you have an implementation idea could describe and I can try to implement.

**Comment by rhmarchiori** [comment]
this is explained here https://github.com/lorenzodifuccia/safaribooks/issues/2

**Comment by rhmarchiori** [comment]
do not forget to check the other issues such as https://github.com/lorenzodifuccia/safaribooks/issues/80, also pointed at https://github.com/lorenzodifuccia/safaribooks/issues/2

**Comment by mustafagonul** [comment]
With my other accounts, cookies.json method is working, but not for the ACM unfortunately on my side.

**Comment by rhm-fyayc** [comment]
I am having issues with my cookies.json for ACM as well. I have tested with my friends cookies.json (also ACM) and it worked. @mustafagonul how old is your acm account? or when did you renewed?

**Comment by rhm-fyayc** [comment]
@matheusfrancisco funcionou para você?

**Comment by mustafagonul** [comment]
I can say that I renewed my membership recently. I can access the books online. But when I try to convert the books to epub, epub is created with headings of the chapters as I could not log in the Safari Books.

**Comment by rhm-fyayc** [comment]
I am in the same scenario then you... I have used the script before in the past. I have renewed my subscription this month and now is not working anymore. Not fully sure that is the reason.. but I'll try to check it more on that topic

**Comment by matheusfrancisco** [comment]
It worked for me the steps i did was...
```
python safaribooks.py --cred "my_user_acm@acm.org:mypass" BOOKID , 
this will generate a log file
```
so I entered the normal oreilly website through the browser and ran this command

```
javascript:(function(){var output = {};document.cookie.split(/\s*;\s*/).forEach(function(pair) {pair = pair.split(/\s*=\s*/);output[pair[0]]=pair.splice(1).join('=');});console.log(JSON.stringify(output));})();
```
I copied all cookies to cookies.json I created
then i use this command and worked well
```
python safaribooks.py BOOK_ID

```

**Comment by mustafagonul** [comment]
I am sure, you can successfully create epub, but on my side, it is not working. :(

As I have said, I have tested with my other account, it is working.

**Comment by rhmarchiori** [comment]
Alright, I have generated a new cookies.json on the safari website and this time it worked. Make sure to be using the last version available on this repo.

**Comment by matheusfrancisco** [comment]
Someone can document this on  README.md ?
That way people can find the solution easier.

**Comment by mustafagonul** [comment]
Please find the logs below:

```
[08/Dec/2019 02:51:35] ** Welcome to SafariBooks! **
[08/Dec/2019 02:51:35] Authentication issue: unable to access profile page.
[08/Dec/2019 02:51:35] Last request done:
	URL: https://learning.oreilly.com/profile/
	DATA: None
	OTHERS: {}

	307
	server: istio-envoy
	cache-control: max-age=0
	content-type: text/html; charset=utf-8
	location: /accounts/login/?next=%2Fprofile%2F
	x-envoy-upstream-service-time: 139
	x-powered-by: Express
	Accept-Ranges: bytes, bytes
	Content-Length: 127
	Date: Sat, 07 Dec 2019 23:51:35 GMT
	Via: 1.1 varnish
	Connection: keep-alive
	X-Client-IP: 217.131.81.131
	X-Served-By: cache-hhn4053-HHN
	X-Cache: MISS
	X-Cache-Hits: 0
	X-Timer: S1575762696.500045,VS0,VE289
	Vary: Accept,Accept, Accept-Encoding, Authorization, Cookie

<p>Temporary Redirect. Redirecting to <a href="/accounts/login/?next=%2Fprofile%2F">/accounts/login/?next=%2Fprofile%2F</a></p>
```

**Comment by matheusfrancisco** [comment]
Try to run ```python safaribooks.py --login BOOK_ID``` will get the error then Get your browser cookies from the learning.oreilly page and place them in cookies.json use the javascript command described above and run ```python safaribooks.py BOOK_ID```

**Comment by mustafagonul** [comment]
I think I have found the problem on my side. When I run `python safaribooks.py --login BOOK_ID` as @matheusfrancisco suggested, cookies.json is created with some stuff. When I check the cookies in the browser, I have found all the fields except **csrftoken**. There was **csrfsafari** which one is the nearest, When I have used the value of csrfsafari for csrftoken, then it worked. :+1:  

Thank you very much :smile:
